llmcam/chat_ui.py

```diff
@@ -122,10 +131,28 @@ def index():
         data_theme="forest", 
         cls="h-[100vh] w-full relative flex flex-row items-stretch overflow-hidden transition-colors z-0 p-0",)
 
-# %% ../nbs/05_chat_ui.ipynb 21
+# %% ../nbs/05_chat_ui.ipynb 22
 # Handle the form submission
 @app.post('/')
-def send(msg: str, contents: list[str] = None, roles: list[str] = None):
+def send(session, msg: str, contents: list[str] = None, roles: list[str] = None):
```

Comment:
`session` can be a class to hold its state? e.g. tools, session_id here?

Reply:
No actually the `session` is referring to FastHTML session. It's a very simple key -> value mapping using Cookies or some other browser -> server information. It should store only a small amount of data so putting `tools` there would not be possible.